components/chart/canvas/Module.tsx

```diff
@@ -19,28 +19,48 @@ function Module({ module, selectModule, isSelected }: ModuleProps) {
   }, []);
 
   return (
-    <div
-      className={`module ${
-        isSelected ? 'active' : ''
-      } ${module.type.toLowerCase()}`}
-      key={module.id}
-      draggable="true"
-      style={{ top: module.y, left: module.x, zIndex: module.id }}
-      onDragStart={(ev) => onDragStart(module.id, ev)}
-      onClick={(e) => {
-        if (e.shiftKey) selectModule(module.id, true);
-        else selectModule(module.id, false);
-      }}
-    >
-      <div className="moduleIcon">
-        <i className={ModuleInfo.getIcon(module.type) + ' moduleIcon '}></i>
-      </div>
-      <div className="moduleNameCon">
-        {moduleName.length >= 34
-          ? moduleName.substring(0, 34) + '...'
-          : moduleName}
-      </div>
-    </div>
+    <>
+      {module.icon ? (
+        <div className={`moduleIconImg ${isSelected ? 'active' : ''
```

Comment:
If we can avoid using css classes and use styled-component would be great
I needed to use CSS classes because I just copied some parts from an old code base

Reply:
done